include/revng/Yield/Internal/Instruction.h

```diff
@@ -0,0 +1,79 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <limits>
+#include <string>
+
+#include "revng/ADT/SortedVector.h"
+#include "revng/Model/VerifyHelper.h"
+#include "revng/Support/MetaAddress.h"
+#include "revng/Support/MetaAddress/YAMLTraits.h"
+#include "revng/Yield/Internal/TagType.h"
+
+namespace yield {
+using ByteContainer = llvm::SmallVector<uint8_t, 16>;
+}
+
+/* TUPLE-TREE-YAML
+
+name: Instruction
+type: struct
+fields:
+  - name: Address
+    type: MetaAddress
+  - name: Bytes
+    type: yield::ByteContainer
+  - name: Raw
+    type: std::string
+
+  - name: Tags
+    sequence:
+      type: std::vector
+      elementType: yield::Tag
+    optional: true
+
+  - name: Opcode
```

Comment:
* Uhm, I feel like *opcode* identifies the number identifying the instruction, while here probably the most appropriate term is *mnemonic*, I think.
* Why do we need this?

Reply:
So this is the name *LLVM* gives to the opcode? Then consider calling it `LLVMOpcode`.
Also, consider that this data structure could be reused with different disassemblers one day.

Reply:
1. yeah, it's llvm notation, sorry for not making that clear.

2. That's actually the reason why I kept it as just an `opcode`, every disassembler should theoretically have something similar after all - hence LLVM in the name could be redundant.

Reply:
`OpcodeIdentifier`? We can decouple this from LLVM and just provide the guarantee that the identifier uniquely identifies the instruction in this architecture.
If you prefer, not to change the name, make sure the doc explains something as the thing above.

Reply:
Sure, I don't really have any attachment to the name.

Reply:
[Done](https://github.com/Cvelth/revng/blob/snapshot/wip_cfg_in_svg/include/revng/Yield/Instruction.h#L40).